os/fs/smartfs/smartfs_utils.c

```diff
@@ -548,6 +548,15 @@ off_t smartfs_seek_internal(struct smartfs_mountpt_s *fs, struct smartfs_ofile_s
 		return sf->filepos;
 	}
 
+	/* Before any further checks/operations, we need to make sure that the length of the file has been calculated */
+
+	if (sf->entry.datalen == SMARTFS_DIRENT_LEN_UNKWN) {
```

Comment:
please test below step, I think this will not work properly.
1. Make 8KB files
2. open created file again.
3. Move file pos to 2kb by lseek
4. Move file pos to 4kb by lseek as same as we called in step #3.

Reply:
Hello, the test seems to work

PSEUDOCODE: -
`open(O_RDWR | O_CREAT)
write(8K)
close()

print_file() //to check contents

open(O_RDONLY)

lseek(2048)
read(fd, buf, 1024)
printf(buf)

lseek(4096)
read(fd, buf, 1024)
printf(buf)
`

```
TASH>>Hello, World!!
************** 8 Kb File Created ****************
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
0000111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
1111222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
2222333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
3333444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
4444555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
5555666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
6666777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
7777
Opened file /mnt/file1 again
Seeked to 2K point
Printing File
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
2222
Seeked to 4K point
Printing File
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
4444
TASH>>
```

Reply:
Hello, the test seems to work

PSEUDOCODE: -
`open(O_RDWR | O_CREAT)
write(8K)
close()

print_file() //to check contents

open(O_RDONLY)

lseek(2048)
read(fd, buf, 1024)
printf(buf)

lseek(4096)
read(fd, buf, 1024)
printf(buf)
`

```
TASH>>Hello, World!!
************** 8 Kb File Created ****************
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
0000111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
111111111111111111111111111111111111111111111111111111111111
1111222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
2222333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
333333333333333333333333333333333333333333333333333333333333
3333444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
4444555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
555555555555555555555555555555555555555555555555555555555555
5555666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
666666666666666666666666666666666666666666666666666666666666
6666777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
777777777777777777777777777777777777777777777777777777777777
7777
Opened file /mnt/file1 again
Seeked to 2K point
Printing File
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
222222222222222222222222222222222222222222222222222222222222
2222
Seeked to 4K point
Printing File
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
444444444444444444444444444444444444444444444444444444444444
4444
TASH>>
```